src/graphql/execution/execute.py

```diff
@@ -998,7 +996,7 @@ def collect_subfields(
         subfields are not repeatedly calculated, which saves overhead when resolving
         lists of values.
         """
-        cache_key = return_type, tuple(field_nodes)
+        cache_key = id(return_type), id(field_nodes)
```

Comment:
The first element of the tuple actually does not need id() to be called, since it uses object identity anyway. It's only needed for the list which is not hashable resp. too slow to hash after conversion to a tuple. Can you change that and adapt the type hint above accordingly?